cli/command/container/diff_test.go

```diff
@@ -0,0 +1,95 @@
+package container
+
+import (
+	"context"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/docker/api/types/container"
+	"github.com/pkg/errors"
+	"gotest.tools/v3/assert"
+)
+
+func TestRunDiff(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		containerDiffFunc: func(
+			ctx context.Context,
+			containerID string,
+		) ([]container.FilesystemChange, error) {
+			return []container.FilesystemChange{
+				{
+					Kind: 0, // Modified ("C")
+					Path: "/path/to/file0",
+				},
+				{
+					Kind: 1, //  Added ("A")
+					Path: "/path/to/file1",
+				},
+				{
+					Kind: 2, // Deleted ("D")
+					Path: "/path/to/file2",
+				},
+			}, nil
+		},
+	})
+
+	cmd := NewDiffCommand(cli)
+	cmd.SetOut(io.Discard)
+
+	containerId := "a61d6b418006"
+	cmd.SetArgs([]string{containerId})
+
+	err := cmd.Execute()
+	assert.Assert(t, err == nil)
+
+	diff := strings.SplitN(cli.OutBuffer().String(), "\n", 3)
+	file0 := strings.TrimSpace(diff[0])
+	file1 := strings.TrimSpace(diff[1])
+	file2 := strings.TrimSpace(diff[2])
+
+	assert.Assert(t, len(diff) == 3)
+	assert.Assert(t, file0 == "C /path/to/file0")
+	assert.Assert(t, file1 == "A /path/to/file1")
+	assert.Assert(t, file2 == "D /path/to/file2")
```

Comment:
Where possible, we try to not fail early; in this case, we can use `assert.Check` instead of `assert.Assert` so that we still show all 3 results, regardless if one of them fails;


```go
assert.Check(t, is.Equal(file0, "C /path/to/file0"))
assert.Check(t, is.Equal(file1, "A /path/to/file1"))
assert.Check(t, is.Equal(file2, "D /path/to/file2"))
```